--- a/jansson-cpp/src/hashtable_seed.cpp
+++ b/jansson-cpp/src/hashtable_seed.cpp
@@ -1,5 +1,7 @@
 /*
 Copyright (c) 2009-2020 Petri Lehtinen <petri@digip.org>
+
+This file is part of Jansson.
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
@@ -16,6 +18,8 @@
 
 #include <stdlib.h>
 #include <time.h>
+#include <sys/time.h>
+#include <unistd.h>
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
--- a/jansson-cpp/src/value.cpp
+++ b/jansson-cpp/src/value.cpp
@@ -30,11 +30,11 @@
 #define JSON_INLINE
 #endif
 
-static JSON_INLINE int isnan(double x) { return x != x; }
-static JSON_INLINE int isinf(double x) { return !isnan(x) && isnan(x - x); }
+static JSON_INLINE int jansson_isnan(double x) { return x != x; }
+static JSON_INLINE int jansson_isinf(double x) { return !jansson_isnan(x) && jansson_isnan(x - x); }
 
 /* json_t */
-static void json_init(json_t *json, json_type type) {
+static void json_init(jansson::json_t *json, jansson::json_type type) {
     json->type = type;
     json->refcount = 1;
 }
--- a/jansson-cpp/src/dump.cpp
+++ b/jansson-cpp/src/dump.cpp
@@ -290,8 +290,7 @@
             size = jsonp_dtostr(buffer, MAX_REAL_STR_LENGTH, value,
                                JSON_REAL_PRECISION(flags));
 #else
-            size = jsonp_dtostr(buffer, MAX_REAL_STR_LENGTH, value,
-                               JSON_REAL_PRECISION(flags));
+            size = jsonp_dtostr(buffer, MAX_REAL_STR_LENGTH, value);
 #endif
             if (size < 0)
                 return -1;
--- a/jansson-cpp/src/pack_unpack.cpp
+++ b/jansson-cpp/src/pack_unpack.cpp
@@ -14,7 +14,7 @@
 
 #include <jansson.h>
 #include <jansson_private.h>
-#include <stdio.h>
+#include <cstdio>
 #include <stdlib.h>
 #include <string.h>
 #include <stdarg.h>
